<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>807</width>
    <height>547</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="font">
   <font>
    <pointsize>9</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Sensor Reading</string>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <property name="sizeGripEnabled">
   <bool>true</bool>
  </property>
  <property name="modal">
   <bool>false</bool>
  </property>
  <layout class="QGridLayout" name="gridLayout" rowstretch="0,1,1,1,1,1" columnstretch="0,3,10,2,2">
   <property name="leftMargin">
    <number>3</number>
   </property>
   <property name="topMargin">
    <number>3</number>
   </property>
   <property name="rightMargin">
    <number>3</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <property name="spacing">
    <number>3</number>
   </property>
   <item row="2" column="2" colspan="3">
    <widget class="QFrame" name="frame_3">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="buttonLayout">
      <property name="spacing">
       <number>2</number>
      </property>
      <property name="leftMargin">
       <number>1</number>
      </property>
      <property name="topMargin">
       <number>1</number>
      </property>
      <property name="rightMargin">
       <number>1</number>
      </property>
      <property name="bottomMargin">
       <number>1</number>
      </property>
     </layout>
    </widget>
   </item>
   <item row="1" column="2" colspan="3">
    <widget class="QFrame" name="frame_2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QVBoxLayout" name="configLayout">
      <property name="spacing">
       <number>2</number>
      </property>
      <property name="leftMargin">
       <number>2</number>
      </property>
      <property name="topMargin">
       <number>2</number>
      </property>
      <property name="rightMargin">
       <number>2</number>
      </property>
      <property name="bottomMargin">
       <number>2</number>
      </property>
     </layout>
    </widget>
   </item>
   <item row="3" column="2" rowspan="3" colspan="3">
    <layout class="QGridLayout" name="gridLayout_2">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <property name="horizontalSpacing">
      <number>2</number>
     </property>
     <property name="verticalSpacing">
      <number>0</number>
     </property>
     <item row="1" column="0" colspan="6">
      <widget class="PlotWidget" name="graph"/>
     </item>
     <item row="2" column="0" colspan="4">
      <widget class="QSlider" name="timebaseSlider">
       <property name="minimum">
        <number>800</number>
       </property>
       <property name="maximum">
        <number>40000</number>
       </property>
       <property name="pageStep">
        <number>100</number>
       </property>
       <property name="value">
        <number>4000</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="2" column="4" colspan="2">
      <widget class="QLabel" name="timebaseLabel">
       <property name="text">
        <string>1000 mS total</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="0" rowspan="6" colspan="2">
    <widget class="QFrame" name="frame">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>10</width>
       <height>10</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777213</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <layout class="QVBoxLayout" name="parameterLayout">
      <property name="leftMargin">
       <number>2</number>
      </property>
      <property name="topMargin">
       <number>2</number>
      </property>
      <property name="rightMargin">
       <number>2</number>
      </property>
      <property name="bottomMargin">
       <number>2</number>
      </property>
      <item>
       <widget class="QLabel" name="label">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Controls</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton">
        <property name="text">
         <string>Initialize</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QListWidget" name="parameterList"/>
      </item>
      <item>
       <widget class="QSpinBox" name="neighbourBox">
        <property name="suffix">
         <string> Neighbours</string>
        </property>
        <property name="prefix">
         <string>+ </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="parameterLabel">
        <property name="text">
         <string>-</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_4">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>120</width>
          <height>25</height>
         </size>
        </property>
        <property name="text">
         <string>CAPTURE DATA</string>
        </property>
        <property name="shortcut">
         <string>C</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="pauseButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Pause When Done</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_2">
        <property name="text">
         <string>SINE FIT</string>
        </property>
        <property name="shortcut">
         <string>S</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_3">
        <property name="text">
         <string>DAMPED SINE FIT</string>
        </property>
        <property name="shortcut">
         <string>D</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_6">
        <property name="text">
         <string>FFT</string>
        </property>
        <property name="shortcut">
         <string>F</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_5">
        <property name="text">
         <string>Auto Y Axis</string>
        </property>
        <property name="icon">
         <iconset resource="res.qrc">
          <normaloff>:/controls/refresh.svg</normaloff>:/controls/refresh.svg</iconset>
        </property>
        <property name="shortcut">
         <string>Y</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="saveButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>25</height>
         </size>
        </property>
        <property name="text">
         <string>Save Data</string>
        </property>
        <property name="shortcut">
         <string>Ctrl+S</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="saveRegionButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>25</height>
         </size>
        </property>
        <property name="text">
         <string>Save Region</string>
        </property>
        <property name="shortcut">
         <string>Ctrl+Shift+S</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QGraphicsView</extends>
   <header>pyqtgraph</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>initialize()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>30</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>sineFit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>140</x>
     <y>366</y>
    </hint>
    <hint type="destinationlabel">
     <x>467</x>
     <y>364</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>dampedSineFit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>140</x>
     <y>395</y>
    </hint>
    <hint type="destinationlabel">
     <x>397</x>
     <y>385</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>restartLogging()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>140</x>
     <y>310</y>
    </hint>
    <hint type="destinationlabel">
     <x>114</x>
     <y>403</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveButton</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>saveTraces()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>137</x>
     <y>511</y>
    </hint>
    <hint type="destinationlabel">
     <x>428</x>
     <y>403</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveRegionButton</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>saveRegion()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>137</x>
     <y>542</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>375</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pauseButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>Dialog</receiver>
   <slot>pauseLogging(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>79</x>
     <y>337</y>
    </hint>
    <hint type="destinationlabel">
     <x>186</x>
     <y>403</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>timebaseSlider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>Dialog</receiver>
   <slot>timebaseChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>374</x>
     <y>545</y>
    </hint>
    <hint type="destinationlabel">
     <x>173</x>
     <y>544</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>autoScaleY()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>137</x>
     <y>480</y>
    </hint>
    <hint type="destinationlabel">
     <x>437</x>
     <y>544</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>parameterList</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>Dialog</receiver>
   <slot>changeParameter(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>52</x>
     <y>166</y>
    </hint>
    <hint type="destinationlabel">
     <x>-8</x>
     <y>170</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>fft()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>26</x>
     <y>424</y>
    </hint>
    <hint type="destinationlabel">
     <x>-4</x>
     <y>354</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>neighbourBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Dialog</receiver>
   <slot>changeNeighbours(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>26</x>
     <y>251</y>
    </hint>
    <hint type="destinationlabel">
     <x>-7</x>
     <y>251</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>changeRange(bool)</slot>
  <slot>initialize()</slot>
  <slot>next()</slot>
  <slot>sineFit()</slot>
  <slot>pauseLogging(bool)</slot>
  <slot>dampedSineFit()</slot>
  <slot>restartLogging()</slot>
  <slot>saveTraces()</slot>
  <slot>setDuration()</slot>
  <slot>saveData()</slot>
  <slot>saveRegion()</slot>
  <slot>timebaseChanged(int)</slot>
  <slot>autoScaleY()</slot>
  <slot>changeParameter(int)</slot>
  <slot>fft()</slot>
  <slot>changeNeighbours(int)</slot>
 </slots>
</ui>
